src/player.rs

```diff
@@ -1,9 +1,21 @@
 use bevy::prelude::*;
+use bevy_rapier2d::prelude::*;
 use bevy_ecs_ldtk::prelude::*;
-
 use crate::physics::{ColliderBundle, GroundDetection};
+use crate::{config::{
+    PLAYER_SETTINGS,
+    GAME_RULES,
+}, game_rules::CoinState};
```

Comment:
Extremely pendantic, but consider newlining CoinState and the ending brace. This probably came from my original mistake anyway.